go-controller/pkg/ovn/controller/admin_network_policy/utils.go

```diff
@@ -0,0 +1,154 @@
+package adminnetworkpolicy
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	addressset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/address_set"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	anpapi "sigs.k8s.io/network-policy-api/apis/v1alpha1"
+)
+
+// getPortProtocol returns the OVN syntax-specific protocol value for a v1.Protocol K8s type
+func getPortProtocol(proto v1.Protocol) string {
+	var protocol string
+	switch proto {
+	case v1.ProtocolTCP:
+		protocol = "tcp"
+	case v1.ProtocolSCTP:
+		protocol = "sctp"
+	case v1.ProtocolUDP:
+		protocol = "udp"
+	}
+	return protocol
+}
+
+// getAdminNetworkPolicyPGName will return the hashed name and provided anp name as the port group name
+func getAdminNetworkPolicyPGName(name string) (hashedPGName, readablePGName string) {
+	return util.HashForOVN(name), name
```

Comment:
portGroup.Name is an index, so this may collide with multicast port group that is based on a namespace name.
I suggest we follow the current format for port groups where the readable name is stored in the `externalIDs["name"]`, and we can add some unique prefix to the name here, like "ANP:" to make sure there are no port groups with the same name.

Reply:
oh good point! lemme see if i can use what you suggested here; I think i already do an owner specific externalID called `ANPExternalIDKey     = "AdminNetworkPolicy"` for the PG but did not consider the naming conflicts that could occur.

Reply:
`portGroup.Name is an index, so this may collide with multicast port group that is based on a namespace name.`
btw the hashForOVN - does it really give the same output for the same input always? ouch.. i thought it would add some random generator stuff...

Reply:
yeah hash is consistent (otherwise we wouldn't be able to update port groups based on a readable name)

Reply:
fixed this one, resolving the comment.